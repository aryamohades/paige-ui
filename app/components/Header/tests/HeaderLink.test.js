import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import HeaderLink from '../HeaderLink';

describe('<HeaderLink />', () => {
  it('should match the snapshot', () => {
    const props = { text: 'test', to: '/' };
    const renderedComponent = renderer
      .create(
        <Router>
          <HeaderLink {...props} />
        </Router>,
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render the correct text', () => {
    const props = { text: 'test', to: '/' };
    const renderedComponent = mount(
      <Router>
        <HeaderLink {...props} />
      </Router>,
    );
    expect(renderedComponent.find(HeaderLink).text()).toEqual(props.text);
  });
});
