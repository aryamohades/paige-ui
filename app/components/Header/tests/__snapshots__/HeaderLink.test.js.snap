// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLink /> should match the snapshot 1`] = `
<a
  className="HeaderLink__StyledLink-kTMJic dLlKZd"
  href="/"
  onClick={[Function]}
>
  <span>
    test
  </span>
</a>
`;
